### ISSUE TITLE
key events not accepted after setting the shortcut

### ISSUE THREAD

**Issue by DivineDominion** [Clipy/KeyHolder#9]
Probably going to fix this myself, but to track the problem until then, here's what I mean:

- click inside a `RecordView`: it obtains key view status and shows a focus ring
- enter any shortcut: it shows the shortcut and the modifiers properly
- try to enter another shortcut

I expected the shortcut to be overwritten, especially since the focus ring is still drawn.

But instead I hear `NSBeep` and the value remains the same until I click inside the view again.

---

The issue is in [`RecordView.endRecording()`](https://github.com/Clipy/KeyHolder/blob/master/Lib/KeyHolder/RecordView.swift#L374):

     if window?.firstResponder == self && !canBecomeKeyView { window?.makeFirstResponder(nil) }

This properly emits an end editing event, but I think it should set the first responder back to `self` afterwards to remain consistent with the visual appearance (i.e. the focus ring that is still there).

**Comment by DivineDominion** [comment]
Turns out that `makeFirstResponder` is not the problem. In the example app, you can reproduce this e.g. by holding shift ⇧ and type a letter, _continue_ holding shift, and type another letter. The second letter will not be accepted.

The problem is that `flagsChanged(_:)` will not be called when the user releases the modifier key, only when pressing it. So we have to reset `inputModifiers` somewhere. 

But resetting it with `inputModifiers = NSEvent.ModifierFlags(rawValue: 0)` in `endRecording` is not a good place. I'll experiment with calling `flagsChanged` from `keyUp` instead, and resetting when key view status is lost.

**Comment by DivineDominion** [comment]
Weirdly, hitting space is configured to always begin recording in `performKeyEquivalent(with:)`. Why is that so?

```swift
        // ...
        } else if Int(theEvent.keyCode) == kVK_Space {
            return beginRecording()
        }
        // ...
```

This allows one to reenable recording:

1. hold shift (and never release)
2. hit a key, e.g. "F" to change the shortcut
3. hit another key, e.g. "D" and notice the beep
4. hit space
5. hit another key again to change the shortcut

I think this is not intentional?

**Comment by Econa77** [comment]
Closed by #11 #47
